pkgs/cli/src/node.zig

```diff
@@ -33,19 +33,159 @@ pub const StartNodeOptions = struct {
     local_priv_key: []const u8,
     logger: *Logger,
 
-    pub fn deinit(self: *StartNodeOptions, allocator: std.mem.Allocator) void {
+    pub fn deinit(self: *NodeOptions, allocator: std.mem.Allocator) void {
         for (self.bootnodes) |b| allocator.free(b);
         allocator.free(self.bootnodes);
         allocator.free(self.validator_indices);
         allocator.free(self.local_priv_key);
     }
 };
 
+/// A Node that encapsulates the networking, blockchain, and validator functionalities.
+/// It manages the event loop, network interface, clock, and beam node.
+pub const Node = struct {
+    loop: xev.Loop,
+    network: networks.EthLibp2p,
+    beam_node: BeamNode,
+    clock: Clock,
+    enr: ENR,
+    options: *const NodeOptions,
+    allocator: std.mem.Allocator,
+
+    const Self = @This();
+
+    pub fn init(self: *Self, allocator: std.mem.Allocator, options: *const NodeOptions) !void {
```

Comment:
this is a weird form of init where one passes the reference to a self which is unitialized

need @gballet 's opinion on this

Reply:
ok, then we need to move a whole lot of inits to this form

Reply:
think it also depends on how large the struct is as the link mentions, if the struct isn't large in size then this form is not necessary.